src/Entries/Collection.php

```diff
@@ -484,6 +485,10 @@ public function save()
             }
         }
 
+        if ($this->isDirty('route')) {
+            Facades\Entry::updateUris($this);
```

Comment:
This doesn't *seem* to be causing any issues when entries are stored as flat files (I don't know where the URIs are being invalidated at the moment).

However, let me know if you'd rather this be done on the Eloquent Driver side and I do that.